app/src/main/java/com/mayokunadeniyi/instantweather/ui/search/SearchFragment.kt

```diff
@@ -70,17 +70,29 @@ class SearchFragment : Fragment() {
         recyclerView.layoutManager = LinearLayoutManager(requireContext())
         recyclerView.adapter = searchResultAdapter
 
-        viewModel.locations.observe(viewLifecycleOwner, Observer { hits ->
+        /* viewModel.locations.observe(viewLifecycleOwner, Observer { hits ->
+             searchResultAdapter.submitList(hits)
+             binding.zeroHits.showIf { hits.size == 0 }
+         })*/
+
+        viewModel.searchHistory.observe(viewLifecycleOwner, Observer { hits ->
+
             searchResultAdapter.submitList(hits)
-            binding.zeroHits.showIf { hits.size == 0 }
+            binding.zeroHits.showIf {
+                hits.isEmpty()
+            }
         })
```

Comment:
Hi, @HayDar-Android this would break the initial behaviour of the code as it would not give the user live suggestions on their location as they type.